src/main/asciidoc/inc/build/_overview.adoc

```diff
@@ -12,9 +12,12 @@ When using this mode, the Dockerfile is created on the fly with all instructions
 Alternatively an external Dockerfile template or Docker archive can be used. This mode is switched on by using one of these three configuration options within
 
 * *dockerFileDir* specifies a directory containing a Dockerfile that will be used to create the image. The name of the Dockerfile is `Dockerfile` by default but can be also set with the option `dockerFile` (see below).
-* *dockerFile* specifies a specific Dockerfile path. The Docker build context directory is set to `dockerFileDir` if given. If not the directory by default is the directory in which the Dockerfile is stored.
```

Comment:
Isn't this behaviour exactly what you wanted to have ? Not sure why you added `contextDir` as I think `dockerFileDir` behaves the same when a `dockerFile` is given. Ok, the naming is not perfect, but that should do the trick, or ?